epoxy-adapter/src/main/java/com/airbnb/epoxy/EpoxyRecyclerView.java

```diff
@@ -507,6 +508,18 @@ public void swapAdapter(Adapter adapter, boolean removeAndRecycleExistingViews)
     clearRemovedAdapterAndCancelRunnable();
   }
 
+  @Override
+  public void onChildAttachedToWindow(@NonNull View child) {
+    if (child instanceof RecyclerView) {
+      RecyclerView recyclerView = (RecyclerView) child;
+      if (recyclerView.getAdapter() instanceof BaseEpoxyAdapter) {
```

Comment:
is `if (recyclerView.getAdapter() instanceof BaseEpoxyAdapter) {` necessary? if the child has had `attach` called on it then it will work, otherwise it is a no-op, right?

Reply:
The idea here was if the child is a `RecyclerView` which is not managed by epoxy there is no point to have visibility tracking. But that's true that the same check is in `EpoxyVisibilityTracker.registerNestedRecyclerView` ... we could remove it from here. This way no need to hookup `onChildAttachedToWindow` method. 

When you say `attach` do you mean `EpoxyVisibilityTracker.attach` ? Because the idea of this implementation was to automatically attach nested `RecyclerView`. May be it is a bad idea and it would be better to have user explicitly call `EpoxyVisibilityTracker.attach` for nested `RecyclerView`s.

Reply:
Actually we could use a `OnChildAttachStateChangeListener` to automatically attach nested visibility trackers from the parent. Not sure what is the best, auto-attach nested or let the the developer attach them ?

Reply:
So as discussed I changed to use `OnChildAttachStateChangeListener` which is simpler.
Thanks @elihart